classes/ActionScheduler_AdminView.php

```diff
@@ -112,6 +113,94 @@ protected function get_list_table() {
 		return $this->list_table;
 	}
 
+	/**
+	 * Action: admin_notices
+	 *
+	 * Maybe check past-due actions, and print notice.
+	 *
+	 * @uses $this->check_pastdue_actions()
+	 */
+	public function maybe_check_pastdue_actions() {
+
+		# Filter to prevent checking actions (ex: inappropriate user).
+		if ( ! apply_filters( 'action_scheduler_check_pastdue_actions', true ) )
+			return;
+
+		# Get last check transient.
+		$last_check = get_transient( 'action_scheduler_last_pastdue_actions_check' );
+
+		# If transient exists, we're within interval, so bail.
+		if ( !empty( $last_check ) )
+			return;
+
+		# Perform the check.
+		$this->check_pastdue_actions();
+	}
+
+	/**
+	 * Check past-due actions, and print notice.
+	 *
+	 * @todo update $link_url to "Past-due" filter when released (see issue #510, PR #511)
+	 */
+	protected function check_pastdue_actions() {
+
+		# Set thresholds.
+		$threshold_seconds = ( int ) apply_filters( 'action_scheduler_pastdue_actions_seconds', DAY_IN_SECONDS );
+		$threshhold_min    = ( int ) apply_filters( 'action_scheduler_pastdue_actions_min', 1 );
+
+		# Allow third-parties to preempt the default check logic.
+		$check = apply_filters( 'action_scheduler_pastdue_actions_check_pre', null );
+
+		# Scheduled actions query arguments.
+		$query_args = array(
+			'date'     => as_get_datetime_object( time() - $threshold_seconds ),
+			'status'   => ActionScheduler_Store::STATUS_PENDING,
+			'per_page' => $threshhold_min,
+		);
+
+		# If no third-party preempted, run default check.
+		if ( is_null( $check ) ) {
```

Comment:
This condition could be changed to another short circuit return and moved to before assigning `$query_args`.

Reply:
@rrennick If a third-party has hooked on to `action_scheduler_pastdue_actions_check_pre` and is using built-in notice, moving this check up would not allow the notice to print.

Reply:
Looking above we can remove this filter and use the `action_scheduler_check_pastdue_actions` to limit where the notice is shown. Sorry for the misdirection.

Reply:
@rrennick It's not clear to me what you're wanting. Can you please explain in more detail?

Reply:
> Can we add a filter to conditionally disable the notice?

This is redundant. The `action_scheduler_check_pastdue_actions` filter can be used to disable the check and notice.

Reply:
@rrennick Ahh gotcha. Yeah, I thought it may be redundant, but then I thought you were differentiating between checking for past-due actions and displaying a notice.

Will remove the filter that disables the notice.

Nothing to do on `L162` then?

Reply:
`action_scheduler_show_pastdue_actions_notice` filter removed in 5d26cc3.

Reply:
> Nothing to do on L162 then?

The `$query_args` array is only used inside the `is_null()` check so it could be moved to that conditional block.

Reply:
@rrennick It’s used later as well, in the action for third-parties to print their own notices.